src/main/java/dev/openfeature/sdk/FeatureProvider.java

```diff
@@ -57,7 +57,11 @@ default void shutdown() {
 
     /**
      * Returns a representation of the current readiness of the provider.
-     * Providers which do not implement this method are assumed to be ready immediately.
+     * If the provider needs to be initialized, it should return {@link ProviderState#NOT_READY}.
+     * If the provider is in an error state, it should return {@link ProviderState#ERROR}.
+     * If the provider is functioning normally, it should return {@link ProviderState#READY}.
+     * 
+     * <p><i>Providers which do not implement this method are assumed to be ready immediately.</i></p>
```

Comment:
At a minimum we should add these docs, but I think the small breaking change below is worth adding.